### PR TITLE
[MPS] Allows scalar input with dim=0 specified

### PR DESCRIPTION
**Pull request by qqaatw**
### Description
Allows reduce ops, scatter, and gather to perform operations like the following:

```
x = torch.tensor(5.0)
out = x.sum(0)

x_mps = x.to("mps")
out = x_mps.sum(0)
```

### CODE REVIEW

**Comment by qqaatw** [comment]
@kulinseth Hi, any chance you can take a look at this PR? Thanks. Also not sure why the PR hasn't been covered by the CLA after signing.

**Comment by kulinseth** [comment]
@qqaatw , added the mps label to run tests.

**Review by kulinseth — COMMENTED** [review]
@qqaatw , can you please 
1. Rebase your change.
2. Add a test case for this in test_mps
3. Make sure the TestConsistency tests in test_mps are passing ?

**Comment by qqaatw** [comment]
@kulinseth Hi, I found this issue has been resolved by #83743 and #85842.

Can you please check with this #82809? I've confirmed that the problem still persists and the PR resolves the problem.
